builtin/credential/oidc/path_config.go

```diff
@@ -0,0 +1,275 @@
+	}
+
+	// Parse issuer_url, required.
+	issuerUrlRaw, ok, err := d.GetOkErr("issuer_url")
+	if ok {
+		_, err = url.Parse(issuerUrlRaw.(string))
+		if err != nil {
+			return logical.ErrorResponse(fmt.Sprintf("Error parsing given issuer_url: %s", err)), nil
+		}
+		cfg.IssuerUrl = issuerUrlRaw.(string)
+	} else if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	} else {
+		return logical.ErrorResponse("auth/oidc: missing required config value issuer_url"), nil
+	}
+	// Parse issuer_verify_ca.
+	issuerVerifyCaRaw, ok, err := d.GetOkErr("issuer_verify_ca")
+	if ok {
+		_, err := certPoolFromString(issuerVerifyCaRaw.(string))
+		if err != nil {
+			return logical.ErrorResponse("auth/oidc: failed parsing PEM of issuer_verify_ca: " + err.Error()), nil
+		}
+		cfg.IssuerCABundle = issuerVerifyCaRaw.(string)
+	} else if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+	// Parse client ids, required.
+	clientIdsRaw, ok, err := d.GetOkErr("client_ids")
+	if ok {
+		cfg.ClientIDs = clientIdsRaw.([]string)
+	} else if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	} else if !ok {
+		return logical.ErrorResponse("auth/oidc: missing required config value client_ids"), nil
+	}
+	// Read defaults for claim configs.
+	cfg.UsernameClaim = d.Get("username_claim").(string)
+	cfg.GroupsClaim = d.Get("groups_claim").(string)
+
+	jsonCfg, err := logical.StorageEntryJSON("config", cfg)
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+	if err := req.Storage.Put(jsonCfg); err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	return nil, nil
+}
+
+func (b *backend) pathConfigExistenceCheck(
+	req *logical.Request, d *framework.FieldData) (bool, error) {
+	cfg, err := b.Config(req.Storage)
+	if err != nil {
+		return false, err
+	}
+
+	return cfg != nil, nil
+}
+
+func (b *backend) oidcProviderForConfig(conf *ConfigEntry) (*oidc.Provider, error) {
+	// TODO(mwitkow): This creates a new Provider each time, making a request to the Discovery URL of the provider. Add caching per config "hash".
+	ctx := context.TODO()
+	if conf.IssuerCABundle != "" {
+		bundles, err := certPoolFromString(conf.IssuerCABundle)
+		if err != nil {
+			return nil, err
+		}
+		transport := &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: bundles,
+			},
+		}
+		httpClient := &http.Client{
+			Transport: transport,
+		}
+		ctx = oidc.ClientContext(ctx, httpClient)
+	}
+
+	return oidc.NewProvider(ctx, conf.IssuerUrl)
+}
+
+func (b *backend) validateAndExtractClaims(cfg *ConfigEntry, rawToken string) (userName string, groups []string, token *oidc.IDToken, err error) {
+	provider, err := b.oidcProviderForConfig(cfg)
+	if err != nil {
+		return "", []string{}, nil, err
+	}
+	if provider == nil {
+		return "", []string{}, nil, errors.New("OIDC provider not configured")
+	}
+	verifier := provider.Verifier(&oidc.Config{SkipClientIDCheck: true}) // note we verify the audience ourselves.
```

Comment:
verifier can also probably just be a field on the backend. Also might just want to pass a custom time.Now for testing.